connor/traderModule.go

```diff
@@ -0,0 +1,539 @@
+func (t *TraderModule) getTokenConfiguration(symbol string, cfg *Config) (float64, float64, float64, error) {
+	switch symbol {
+	case "ETH":
+		return cfg.ChargeIntervalETH.Start, cfg.ChargeIntervalETH.Destination, cfg.Distances.StepForETH, nil
+	case "ZEC":
+		return cfg.ChargeIntervalZEC.Start, cfg.ChargeIntervalZEC.Destination, cfg.Distances.StepForZEC, nil
+	case "XMR":
+		return cfg.ChargeIntervalXMR.Start, cfg.ChargeIntervalXMR.Destination, cfg.Distances.StepForXMR, nil
+	}
+	return 0, 0, 0, nil
+}
+
+func (t *TraderModule) ChargeOrdersOnce(ctx context.Context, symbol string, token watchers.TokenWatcher, snm watchers.PriceWatcher, balanceReply *sonm.BalanceReply) error {
+	t.c.db.CreateOrderDB()
+	start, destination, step, err := t.getTokenConfiguration(symbol, t.c.cfg)
+	if err != nil {
+		log.Printf("cannot get token configuraton %v", err)
+		return err
+	}
+
+	count, err := t.c.db.GetCountFromDB()
+	if err != nil {
+		log.Printf("Cannot get count from DB: %v", err)
+		return err
+	}
+	if count == 0 {
+		log.Printf("Save TEST order cause DB is empty! \r\n")
+		if err := t.c.db.SaveOrderIntoDB(&database.OrderDb{
+			OrderID:         0,
+			Price:           0,
+			Hashrate:        0,
+			StartTime:       time.Time{},
+			ButterflyEffect: 2,
+			ActualStep:      start,
+		}); err != nil {
+			fmt.Printf("Cannot save order into DB %v\r\n", err)
+		}
+	}
+
+	pricePerMonthUSD, pricePerSecMh, err := t.GetPriceForTokenPerSec(token, symbol)
+	if err != nil {
+		log.Printf("Cannot get profit for tokens: %v\r\n", err)
+		return err
+	}
+	limitChargeInSNM := t.profit.LimitChargeSNM(balanceReply.GetSideBalance().Unwrap(), t.c.cfg.Sensitivity.PartCharge)
+	limitChargeInSNMClone := big.NewInt(0).Set(limitChargeInSNM)
+	limitChargeInUSD := t.profit.ConvertingToUSDBalance(limitChargeInSNMClone, snm.GetPrice())
+
+	mhashForToken, err := t.c.db.GetLastActualStepFromDb()
+	if err != nil {
+		log.Printf("Cannot get last actual step from DB %v\r\n", err)
+		return err
+	}
+
+	pricePackMhInUSDPerMonth := mhashForToken * (pricePerMonthUSD * t.c.cfg.Sensitivity.MarginAccounting)
+	sumOrdersPerMonth := limitChargeInUSD / pricePackMhInUSDPerMonth
+	if limitChargeInSNM.Int64() == 0 {
+		log.Printf("Balance SNM is not enough for create orders! %v", limitChargeInSNM.Int64())
+		return err
+	}
+
+	t.c.logger.Info("i", zap.String("CHARGE ORDERS ONCE: ", symbol),
+		zap.String("Limit for Charge SNM :", t.PriceToString(limitChargeInSNM)),
+		zap.Float64("Limit for Charge USD :", limitChargeInUSD),
+		zap.Float64("Pack Price per month USD", pricePackMhInUSDPerMonth),
+		zap.Int("Sum orders per month", int(sumOrdersPerMonth)))
+
+	for i := 0; i < int(sumOrdersPerMonth); i++ {
+		if mhashForToken >= destination {
+			log.Printf("Charge is finished cause reached the limit %.2f Mh/s\r\n", t.c.cfg.ChargeIntervalETH.Destination)
+			break
+		}
+		pricePerSecPack := t.FloatToBigInt(mhashForToken * pricePerSecMh)
+		zap.String("Price :: %v\r\n", t.PriceToString(pricePerSecPack))
```

Comment:
that not how zap works...

Reply:
zap logger is logging isn't it?) Zap works.

Reply:
Oh, u mean use ONLY one type of logger,Ok. I know, haven't finished yet.

Reply:
`zap.String` does not log anything, please refer to the documentation or grep by our repository.